pallets/pool-system/src/impls.rs

```diff
@@ -367,6 +372,60 @@ impl<T: Config> InvestmentAccountant<T::AccountId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> ChangeGuard for Pallet<T> {
+	type Change = T::RuntimeChange;
+	type ChangeId = T::Hash;
+	type PoolId = T::PoolId;
+
+	fn note(pool_id: Self::PoolId, change: Self::Change) -> Result<Self::ChangeId, DispatchError> {
+		let registered_change = (Self::now(), change.clone());
+		let change_id: Self::ChangeId = T::Hashing::hash(&registered_change.encode());
```

Comment:
Yes (confirmed previously with @offerijns), because the hash also takes into account the change content. So, if in the same block, we have the exact change, it will count as only one change. Make sense for all use cases that came to my mind

Reply:
Would only be the case if there are multiple updates that have exactly the same content (same change it's applying), right? So I think thats' fine

Reply:
Since the contents of the change are included in the hash

Reply:
Ahh yeah.